src/main/src/main/java/org/geoserver/ows/ProxifyingURLMangler.java

```diff
@@ -95,10 +95,15 @@ public void mangleURL(
 
         // Mangles the URL base in different ways based on a flag
         // (for two reasons: a) speed; b) to make the admin aware of
-
         // possible security liabilities)
 
-        if (this.geoServer.getGlobal().isUseHeadersProxyURL() == true && proxyBase != null) {
+        boolean workspaceEnabled = this.geoServer.getSettings().getWorkspace() != null;
+        boolean doMangleHeaders =
+                workspaceEnabled
+                        ? this.geoServer.getSettings().isUseHeadersProxyURL()
+                        : this.geoServer.getGlobal().isUseHeadersProxyURL();
+
+        if (proxyBase != null && doMangleHeaders) {
```

Comment:
Please add some FINE level logging to know whats happening and the result of this logic conditions execution.

Reply:
While it seems ok to set the flag in `SettingsInfo`, there's still an impedance mismatch with `GeoServer.getGlobal().getSettings():SeettingsInfo`, on which these would be of no use.
If `isUseHeadersProxyURL` is **moved** instead to `SettingsInfo`, everything would be consistent and this decision making unnecessary, leaving just `doMangleHeaders = this.geoServer.getSettings().isUseHeadersProxyURL()`, and letting `geoServer.getSettings()` decide whether to return the workspace or the global settings.

That of course implies deprecating `GeoServerInfo.is/setUseHeadersProxyURL` and defer to its internal `SettingsInfo` (e.g. `@Deprecated public Boolean isUseHeadersProxyURL() { return getSettings().isUseHeadersProxyURL();}`)

Reply:
@aaime thoughts?

Reply:
hmm thinking further, still the workspace settings may not have the value set at all and you'll still need to manually select the global settings value. For which you would need to define it as a `Boolean` instead of `boolean` and allow it to be `null`. Maybe add an `Optional<Boolean> useUseHeadersProxyURL(){return Optional.ofNullable(this.useHeadersProxyURL);}`